wgpu-core/src/instance.rs

```diff
@@ -177,10 +177,16 @@ impl Surface {
                 .ok_or(GetSurfacePreferredFormatError::UnsupportedQueueFamily)?
         };
 
-        preferred_formats
+        let supported_formats = candidate_formats
             .iter()
             .cloned()
-            .find(|preferred| caps.formats.contains(preferred))
+            .filter(|candidate| caps.formats.contains(candidate))
```

Comment:
Instead of filtering the candidates we should just read the capabilities and return the formats the device says its capable of